hydra/core/callbacks.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import warnings
+from typing import Any, List
+
+from omegaconf import DictConfig
+
+from hydra._internal.callback import Callback
+from hydra.core.singleton import Singleton
+from hydra.core.utils import JobReturn
+from hydra.utils import instantiate
+
+
+class Callbacks(metaclass=Singleton):
+    @staticmethod
+    def instance(*args: Any, **kwargs: Any) -> "Callbacks":
+        ret = Singleton.instance(Callbacks, *args, **kwargs)
+        assert isinstance(ret, Callbacks)
+        return ret
+
+    def __init__(self) -> None:
+        self.callbacks: List[Callback] = []
+
+    def instantiate_callbacks(self, config: DictConfig) -> None:
+        if config.hydra.callbacks is None:
+            raise RuntimeError("Hydra Callbacks is not configured")
+
+        assert len(self.callbacks) == 0, "Callbacks already instantiated."
+
+        for params in config.hydra.callbacks.values():
+            self.callbacks.append(instantiate(params))
+
+    def _iterate(
```

Comment:
notify or fireEvent sounds better.
```suggestion
    def _notify(
```

Reply:
fixed!